api/client/diff.go

```diff
@@ -19,11 +19,8 @@ import (
 func (cli *DockerCli) CmdDiff(args ...string) error {
 	cmd := cli.Subcmd("diff", []string{"CONTAINER"}, "Inspect changes on a container's filesystem", true)
 	cmd.Require(flag.Exact, 1)
-	cmd.ParseFlags(args, true)
 
-	if cmd.Arg(0) == "" {
```

Comment:
Hi @duglin. I think you are right.
That's my fault. Will fix and change the title